### PR TITLE
[LF] move TextMap to 2.dev

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**

### CODE REVIEW

File: daml-lf/archive/src/main/protobuf/com/daml/daml_lf_dev/daml_lf2.proto [hunk 1]
```diff
@@ -355,18 +350,26 @@ enum BuiltinFunction {
...
-  BIGNUMERIC_TO_TEXT = 1010;           // *Available in versions >= 2.dev
-
-  TYPE_REP_TYCON_NAME = 1011;          // *Available in versions >= 2.dev*
+
+  TEXTMAP_EMPTY = 1001;
+  TEXTMAP_INSERT = 1002;
+  TEXTMAP_LOOKUP = 1003;
+  TEXTMAP_DELETE = 1004;
+  TEXTMAP_TO_LIST = 1005;
+  TEXTMAP_SIZE = 1006;
```
Comment (paulbrauner-da) [inline daml-lf/archive/src/main/protobuf/com/daml/daml_lf_dev/daml_lf2.proto:359]: Should it also say available in >= 2.dev?

File: daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/TransactionVersion.scala [hunk 2]
```diff
@@ -44,7 +44,8 @@ object TransactionVersion {
   def maxVersion: TransactionVersion = VDev
 
   // TODO(https://github.com/digital-asset/daml/issues/18240) move this feature flag to VDev.
-  private[lf] val minContractKeys = V31
+  private[lf] val minContractKeys = VDev
```
Comment (paulbrauner-da) [inline daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/TransactionVersion.scala:47]: Leftover from testing? (Done in https://github.com/digital-asset/daml/pull/18751)

File: language-support/ts/codegen/tests/ts/build-and-lint-test/src/__tests__/test.ts [hunk 3]
```diff
@@ -31,6 +31,7 @@ import * as buildAndLint from "@daml.js/build-and-lint-1.0.0";
 const LEDGER_ID = "build-and-lint-test";
 const APPLICATION_ID = "build-and-lint-test";
 const SECRET_KEY = "secret";
+const DAML_TEXTMAP = true;
```
Comment (paulbrauner-da) [inline language-support/ts/codegen/tests/ts/build-and-lint-test/src/__tests__/test.ts:34]: I suppose that this maybe be false at some point if you introduced it?

**Review by paulbrauner-da — APPROVED** [review]
Thanks!

File: daml-lf/archive/src/main/protobuf/com/daml/daml_lf_dev/daml_lf2.proto (hunk 1 above)
Comment (remyhaemmerle-da) [reply daml-lf/archive/src/main/protobuf/com/daml/daml_lf_dev/daml_lf2.proto:359]: Yes.

File: compiler/damlc/daml-stdlib-src/DA/TextMap.daml [hunk 4]
```diff
@@ -1,6 +1,14 @@
 -- Copyright (c) 2024 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
 -- SPDX-License-Identifier: Apache-2.0
 
+{-# LANGUAGE CPP #-}
+
+#ifndef DAML_BIGNUMERIC
```
Comment (moisesackerman-da) [inline compiler/damlc/daml-stdlib-src/DA/TextMap.daml:6]:
```suggestion
#ifndef DAML_TEXTMAP
```

File: daml-lf/language/src/main/scala/com/digitalasset/daml/lf/language/LanguageVersion.scala [hunk 5]
```diff
@@ -52,6 +52,7 @@ object LanguageVersion {
     val extendedInterfaces = v2_dev
 
     /** BigNumeric */
+    val textMap = v2_dev
     val bigNumeric = v2_dev
```
Comment (moisesackerman-da) [inline daml-lf/language/src/main/scala/com/digitalasset/daml/lf/language/LanguageVersion.scala:56]:
```suggestion
    /** TextMap */
    val textMap = v2_dev
    /** BigNumeric */
    val bigNumeric = v2_dev
```

File: language-support/hs/bindings/test/daml/for-tests/Valuepedia.daml [hunk 6]
```diff
@@ -28,7 +29,11 @@ data Bucket = Bucket
     , unit : ()
     , date : Date
     , opts : [Optional Text]
-    , map : TextMap Int
+#ifndef DAML_TEXMAP
+    , textMap : TextMap Int
+#else 
+    , textMap : Unit
```
Comment (moisesackerman-da) [inline language-support/hs/bindings/test/daml/for-tests/Valuepedia.daml:35]: do we need the `else` case?

File: language-support/java/codegen/src/it/daml/Lib_2_1.daml [hunk 7]
```diff
@@ -14,7 +14,5 @@ import Tests.NumericTest()
 
 import Tests.Import()
 import Tests.Escape()
-import Tests.TextMapTest()
-import Tests.GenMapTest()
```
Comment (moisesackerman-da) [inline language-support/java/codegen/src/it/daml/Lib_2_1.daml:18]: don't we want to keep `Tests.GenMapTest()`?

File: daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/value/ValueSpec.scala [hunk 8]
```diff
@@ -32,7 +32,7 @@ class ValueSpec
     with ScalaCheckPropertyChecks {
   import ValueSpec._
 
-  "ContractID.V1.build" - {
+  "ContractID.V3.build" - {
```
Comment (moisesackerman-da) [inline daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/value/ValueSpec.scala:35]: is this change needed?
